Remove `polyfill.io`

SukkaW opened the pull request:
TypeScript website loads polyfill from `polyfill.io` inside the `<Layout />`. The PR replaces `polyfill.io` with `cdnjs.cloudflare.com/polyfill`.

----

`polyfill.io` was acquired by **a China-based CDN company** "Funnull", see [the announcement from the `polyfill.io` domain owner's Twitter](https://x.com/JakeDChampion/status/1761315227008643367) and https://github.com/polyfillpolyfill/polyfill-service/issues/2834. Despite Funnull's claims of operating in the United States, the predominance of Simplified Chinese on its website suggests otherwise, and it turns out that **"Funnull" is notorious for providing service for the betting and pornography industries**.

[The original creator of the `polyfill.io` has voiced his concern on Twitter](https://twitter.com/triblondon/status/1761852117579427975). And since the acquisition, numerous issues have emerged (https://github.com/polyfillpolyfill/polyfill-service/issues/2835, https://github.com/polyfillpolyfill/polyfill-service/issues/2838, https://github.com/alist-org/alist/issues/6100), rendering the `polyfill.io` service **extremely unstable**. Since then, Fastly ([Announcement](https://community.fastly.com/t/new-options-for-polyfill-io-users/2540)) and Cloudflare ([Announcement](https://blog.cloudflare.com/polyfill-io-now-available-on-cdnjs-reduce-your-supply-chain-risk)) has hosted their own instances of `polyfill.io` service.

jakebailey commented:
Can we just remove the polyfill.io reference entirely? It's 2024; all modern browsers are evergreen.

SukkaW commented:
> Can we just remove the polyfill.io reference entirely? It's 2024; all modern browsers are evergreen.

Don't know why `polyfill.io` was even introduced back then. But since it is 2024 already, let's assume all browsers now have ES6 support. I will rebase my PR.

----

@jakebailey I've rebased my PR and `polyfill.io` usage is now removed.
